### ISSUE TITLE
Error on Android : Attempt to invoke virtual method 

### ISSUE THREAD

**Issue by LaurelyneD** [liuminghui2017/react-native-esptouch#1]
Hello, 
Thanks for this module! 

I get an error on Android when i call " RNEsptouch.initESPTouch();": 

` attempt to invoke virtual method int android content.Context.checkPermission(java.lang.String, int, int) on a null object reference`

So,  i made myself the permission with React-native-permission  and  i do a quick change on the code

`> node_modules > react-native-epstouch > android > src > main > java > ... > RNEsptouchModule.java`

@ReactMethod

    public void initESPTouch() {
        if (isSDKAtLeastP()) {

            if (ContextCompat.checkSelfPermission(thisActivity, Manifest.permission.ACCESS_COARSE_LOCATION)
                    != PackageManager.PERMISSION_GRANTED) {

                if (ActivityCompat.shouldShowRequestPermissionRationale(thisActivity,
                        Manifest.permission.ACCESS_COARSE_LOCATION)) {
                    Toast.makeText(thisActivity, "ESPTouch配置需要此权限", Toast.LENGTH_LONG);
                }
                String[] permissions = {Manifest.permission.ACCESS_COARSE_LOCATION};
                ActivityCompat.requestPermissions(thisActivity, permissions, REQUEST_PERMISSION);

            } else {
                registerBroadcastReceiver();
            }
        } else {
            registerBroadcastReceiver();
        }
    }
         

To 

    @ReactMethod

    public void initESPTouch() {

         registerBroadcastReceiver();
         }}

or 

 It's works well! can  you do the modification of something that hide this error ?
 Thanks !

**Comment by joelzavala** [comment]
Hi @LyneCaat.
Did it work for you? I am testing it with the configuration you made but the application closes.

Did you do anything else besides this change? Thanks in advance : )

**Comment by vilelalabs** [comment]
> Hi @joelzavala 

Try to check the "Android permissions" on Manifest file (on **android\app\src\main\AndroidManifest.xml**).

 I added these - maybe don't need add all of them:
```
  <uses-permission android:name="android.permission.INTERNET" />
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
  <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
  <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
  <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
And it works immediately on android emulator. 
For test on device, you'll need to add manually in App Info->Permissions after installed. But I think it's easy to solve this with RN (trying to do right now).

**Comment by joelzavala** [comment]
Hi @hvilela0
App info is app.json file in native react project? If not, where do I find this file?
Thank you very much in advance :)

**Comment by vilelalabs** [comment]
Hi @joelzavala

No, App info ℹ  is an option for an app installed on a Android device (don't know how ir works on iOS).
Appears when you hold the app icon on app selection screen in your smartphone.
You even can find this on **Settings-> Apps and Notifications** I think.

(I'm trying to understand RN, Java, Javascript and this library.. so I'm no tottaly right if my solution will work for you )

You're welcome.

**Comment by joelzavala** [comment]
Hello @hvilela0 

I added the permissions that you mentioned earlier but it keeps throwing me the same problem, which is:
Try invoking virtual method "int android.content.Context.checkPermission (java.lang.String, int, int) on a null object reference

**Comment by vilelalabs** [comment]
Hi @joelzavala 

You are right that you edit the right Manifest file? I had issues because the search from VS Code (I don't know if your're using it) returned me the wrong Manifest file (same name, but wrong directory) check out this. (the right directory is: on android\app\src\main\AndroidManifest.xml) as I said before.

And remember, always you changed the .java file and/or manifest.xml, rebuild the application with ```react-native run-android``` command.

If it's right, sorry maybe I can't help you.

**Comment by joelzavala** [comment]
Hi @hvilela0 
The library has already worked for you?

**Comment by vilelalabs** [comment]
> Hi @joelzavala 

Yes! Works fine after modifications suggested by @LyneCaat  and permissions added.

**Comment by joelzavala** [comment]
Hi @hvilela0 

What version of Android do you use and what version of react native?
Is it because I am using a new version of react native?

**Comment by vilelalabs** [comment]
Hi @joelzavala 
Android 10 & RN 5.0.1
I really don't understand so much about RN versions and differences (I'm very new on React / RN and App programming .. learning yet)
